datafusion/core/tests/physical_optimizer/push_down_filter.rs

```diff
@@ -154,29 +153,25 @@ impl FileSource for TestSource {
 
     fn try_pushdown_filters(
         &self,
-        mut fd: FilterDescription,
+        filters: &[Arc<dyn PhysicalExpr>],
         config: &ConfigOptions,
-    ) -> Result<FilterPushdownResult<Arc<dyn FileSource>>> {
+    ) -> Result<FilterPushdownPropagation<Arc<dyn FileSource>>> {
+        let mut all_filters = filters.iter().map(Arc::clone).collect::<Vec<_>>();
```

Comment:
For example, if this was in a struct, we could implement `clone()` for the struct and hide this complexity